### PR TITLE
Added local_folder param to get_result_df (#394)

### PR DESCRIPTION
**Pull request by shivamsanju**
**This PR addresses issue #394 :**
Previously users were not getting an option to download the result dataframe on the local directory because a temporary directory was getting created which was cleaned up after reading the dataframe.

**Changes:**
Now an optional parameter called local_folder is added where users can pass the local directory name where they want to download the result. If the user does not pass this parameter, the function will create a temporary directory and clean it after reading the dataframe.

### CODE REVIEW

**Comment by xiaoyongzhu** [comment]
@shivamsanju thanks for the contribution! Overall this looks good to me. I put a few comments inline.

File: feathr_project/feathr/utils/job_utils.py [hunk 1]
```diff
@@ -7,28 +7,34 @@
 import tempfile
 
 
-def get_result_df(client: FeathrClient, format: str = None, res_url: str = None) -> pd.DataFrame:
+def get_result_df(client: FeathrClient, format: str = None, res_url: str = None, local_folder: str = None) -> pd.DataFrame:
     """Download the job result dataset from cloud as a Pandas dataframe.
 
     format: format override, could be "parquet", "delta", etc.
     res_url: output URL to download files. Note that this will not block the job so you need to make sure the job is finished and result URL contains actual data.
```
Comment (xiaoyongzhu) [inline feathr_project/feathr/utils/job_utils.py:14]: Add the pydoc string for local_folder

File: feathr_project/feathr/utils/job_utils.py [hunk 2]
```diff
@@ -7,28 +7,34 @@
...
     """
     res_url: str = res_url or client.get_job_result_uri(block=True, timeout_sec=1200)
     format: str = format or client.get_job_tags().get(OUTPUT_FORMAT, "")
-    tmp_dir = tempfile.TemporaryDirectory()
-    client.feathr_spark_launcher.download_result(result_path=res_url, local_folder=tmp_dir.name)
+    # if local_folder params is not provided then create a temporary folder
+    if local_folder is not None:
+        local_dir_path = os.getcwd() + local_folder
```
Comment (xiaoyongzhu) [inline feathr_project/feathr/utils/job_utils.py:20]: Maybe we just use absolute path rather than a local_folder under `os.getcwd()`?

**Comment by xiaoyongzhu** [comment]
@shivamsanju can you merge the main branch of Feathr into this PR? Seems like those are the root cause for those test failures

**Comment by shivamsanju** [comment]
> @shivamsanju can you merge the main branch of Feathr into this PR? Seems like those are the root cause for those test failures

Hi @xiaoyongzhu, I have already merged the main branch of linkedin/feathr to this PR (commit b2ee907267103bf7d24ddfe4e6d1754b79fa54db). It says already up to date when I do a git pull from linkedin/feathr.

File: feathr_project/feathr/utils/job_utils.py [hunk 3]
```diff
@@ -7,28 +7,35 @@
...
-    tmp_dir = tempfile.TemporaryDirectory()
-    client.feathr_spark_launcher.download_result(result_path=res_url, local_folder=tmp_dir.name)
+    # if local_folder params is not provided then create a temporary folder
+    if local_folder is not None:
+        local_dir_path = local_folder
+    else:
+        tmp_dir = tempfile.TemporaryDirectory()
+        local_dir_path = tmp_dir.name
+        
+    client.feathr_spark_laucher.download_result(result_path=res_url, local_folder=local_dir_path)
```
Comment (xiaoyongzhu) [inline feathr_project/feathr/utils/job_utils.py:26]: client.feathr_spark_launcher.download_result

**Comment by xiaoyongzhu** [comment]
> > @shivamsanju can you merge the main branch of Feathr into this PR? Seems like those are the root cause for those test failures
> 
> Hi @xiaoyongzhu, I have already merged the main branch of linkedin/feathr to this PR (commit [b2ee907](https://github.com/linkedin/feathr/commit/b2ee907267103bf7d24ddfe4e6d1754b79fa54db)). It says already up to date when I do a git pull from linkedin/feathr.

Ah OK so it might be a small typo that I've pointed out (`feathr_spark_laucher` to `feathr_spark_launcher`)

**Review by xiaoyongzhu — APPROVED** [review]

File: feathr_project/feathr/utils/job_utils.py [hunk 4]
```diff
@@ -7,28 +7,35 @@
 import tempfile
 
 
-def get_result_df(client: FeathrClient, format: str = None, res_url: str = None) -> pd.DataFrame:
+def get_result_df(client: FeathrClient, format: str = None, res_url: str = None, local_folder: str = None) -> pd.DataFrame:
     """Download the job result dataset from cloud as a Pandas dataframe.
 
     format: format override, could be "parquet", "delta", etc.
     res_url: output URL to download files. Note that this will not block the job so you need to make sure the job is finished and result URL contains actual data.
+    local_folder: optional parameter to specify the absolute download path. if the user does not provide this, function will create a temporary directory and delete it after reading the dataframe.
```
Comment (hangfei) [inline feathr_project/feathr/utils/job_utils.py:15]: nit: can it also be relative? If both, will give it an example better to explain this?

Comment (xiaoyongzhu) [reply feathr_project/feathr/utils/job_utils.py:15]: @hangfei I thought about whether to use relative path or absolute path, and feel absolute path should able to cover most of the use cases (actually @shivamsanju 's original implementation is for relative path and I asked him to change to absolute path).

**Pull request merged**